src/parser.cc

```diff
@@ -1074,11 +1074,18 @@ void addNestedModel(ElementPtr _sdf, ElementPtr _includeSDF)
   std::string modelName = modelPtr->Get<std::string>("name");
   while (elem)
   {
-    if (elem->GetName() == "link")
+    // protect elements that should not change the name attribute
+    // i.e: plugin
+    if ((elem->GetName() == "link") || (elem->GetName() == "joint") ||
+        (elem->GetName() == "model"))
     {
       std::string elemName = elem->Get<std::string>("name");
       std::string newName =  modelName + "::" + elemName;
       replace[elemName] = newName;
+    }
+
+    if (elem->GetName() == "link")
```

Comment:
Quick note that this will be in conflict with #596, but we'll definitely need the line from there.

Reply:
+1 going to merge both, I'll keep an eye on it, tests should fail we miss it.